types/transaction.go

```diff
@@ -28,16 +63,17 @@ func (t *Transaction) IsContractCreation() bool {
 	return t.To == nil
 }
 
+func (t *Transaction) IsTypedTransaction() bool {
+	return t.Type != TxTypeLegacy
+}
+
 // ComputeHash computes the hash of the transaction
 func (t *Transaction) ComputeHash() *Transaction {
-	ar := marshalArenaPool.Get()
 	hash := keccak.DefaultKeccakPool.Get()
-
-	v := t.MarshalRLPWith(ar)
-	hash.WriteRlp(t.Hash[:0], v)
-
-	marshalArenaPool.Put(ar)
-	keccak.DefaultKeccakPool.Put(hash)
+	if _, err := hash.Write(t.MarshalRLP()); err == nil {
+		hash.Sum(t.Hash[:0])
+		keccak.DefaultKeccakPool.Put(hash)
+	}
 
 	return t
```

Comment:
If the hashing fails for whatever reason, we just return the transaction without the hash filled?

Where do we even set the transaction hash here?

Reply:
@zivkovicmilos 

> If the hashing fails for whatever reason, we just return the transaction without the hash filled?
Yes, we should return error also?

> Where do we even set the transaction hash here?
t.Hash

Reply:
🙂 
I know that the hash is supposed to be in `t.Hash`, I'm asking at which point we actually set this field here.

Is `ComputeHash` an old function that was already there?

Reply:
@zivkovicmilos 

>  I'm asking at which point we actually set this field here.
hash.Sum(t.Hash[:0])

> Is ComputeHash an old function that was already there?
Yes

Reply:
Got it 👍 

Let's leave this for now, since it's a legacy func